src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCacheSpec.scala

```diff
@@ -1,142 +1,50 @@
-    (actorRef ? GetByProjectAndName(project, name3)).futureValue shouldBe ClusterNotFound
-
-    // the cycle still eventually populates normally
-
     eventually {
-      (actorRef ? GetByProjectAndName(project, name1)).futureValue shouldBe ClusterReady(Host(s"${cluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org"))
-      (actorRef ? GetByProjectAndName(project, name2)).futureValue shouldBe ClusterNotReady
-      (actorRef ? GetByProjectAndName(project, name3)).futureValue shouldBe ClusterPaused
+      clusterDnsCache.projectNameToHost.get(DnsCacheKey(cluster2.googleProject, cluster2.clusterName)).futureValue shouldEqual
+        HostNotReady
     }
 
-    val newName1 = ClusterName("newname1")
-    val newName2 = ClusterName("newname2")
-
-    val newCluster1 = cluster1.copy(clusterName = newName1, dataprocInfo = cluster1.dataprocInfo.copy(hostIp = Some(IP("a new IP")), googleId = Option(UUID.randomUUID())))
-    val newCluster2 = cluster2.copy(clusterName = newName2, dataprocInfo = cluster2.dataprocInfo.copy(hostIp = Some(IP("another new IP")), googleId = Option(UUID.randomUUID())))
-
-    newCluster1.save() shouldEqual newCluster1
-    newCluster2.save() shouldEqual newCluster2
-
-    // add one cluster to cache immediately without waiting for the cycle
-
-    actorRef ! ProcessReadyCluster(newCluster1)
-
-    // the new cluster is available in the DNS cache (as well as the older clusters) but the other is not
-
-    (actorRef ? GetByProjectAndName(project, name1)).futureValue shouldBe ClusterReady(Host(s"${cluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org"))
-    (actorRef ? GetByProjectAndName(project, name2)).futureValue shouldBe ClusterNotReady
-    (actorRef ? GetByProjectAndName(project, newName1)).futureValue shouldBe ClusterReady(Host(s"${newCluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org"))
-    (actorRef ? GetByProjectAndName(project, newName2)).futureValue shouldBe ClusterNotFound
-
-    // the cycle still eventually populates normally
-
-    eventually {
-      (actorRef ? GetByProjectAndName(project, name1)).futureValue shouldBe ClusterReady(Host(s"${cluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org"))
-      (actorRef ? GetByProjectAndName(project, name2)).futureValue shouldBe ClusterNotReady
-      (actorRef ? GetByProjectAndName(project, newName1)).futureValue shouldBe ClusterReady(Host(s"${newCluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org"))
-      (actorRef ? GetByProjectAndName(project, newName2)).futureValue shouldBe ClusterReady(Host(s"${newCluster2.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org"))
-    }
+    ClusterDnsCache.HostToIp.get(Host(s"${cluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org")) shouldBe cluster1.dataprocInfo.hostIp
 
+    ClusterDnsCache.HostToIp.get(Host(s"${cluster2.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org")) shouldBe None
```

Comment:
Can we add another test with a cluster that `isStartable` and verify that its `HostStatus` ends up being `HostPaused` and its `ClusterDnsCache` entry is `None`?